src/components/structures/RoomDirectory.tsx

```diff
@@ -150,8 +153,8 @@ export default class RoomDirectory extends React.Component<IProps, IState> {
             publicRooms: [],
             loading: true,
             error: null,
-            instanceId: undefined,
-            roomServer: MatrixClientPeg.getHomeserverName(),
+            instanceId: localStorage.getItem(LAST_INSTANCE_KEY),
+            roomServer: localStorage.getItem(LAST_SERVER_KEY) || MatrixClientPeg.getHomeserverName(),
```

Comment:
what happens if that roomServer is no longer defined? e.g it was in config.json but is no longer there
what happens if that instanceId is no longer a valid one offered by the HS?